index.js

```diff
@@ -1 +1,319 @@
+    }
+    console.log(result);
+  });
+  res.send('working on it');
+}
+
+function renderSingleSpecies(req, res) {
+  const id = req.params.index;
+  const singleSpeciesQuery = `SELECT * FROM species WHERE id = '${id}'`;
+  pool.query(singleSpeciesQuery, (err, result) => {
+    if (err) {
+      console.log(err);
+      return err;
+    }
+    console.log(result.rows);
+    const notesSpeciesQuery = `SELECT notes.habitat, notes.flock_size, notes.date, notes.time, notes.day, notes.behaviour, species.name, species.scientific_name FROM notes INNER JOIN species ON species.id= notes.species_id WHERE species_id = '${id}'`;
+
+    pool.query(notesSpeciesQuery, (innerJoinErr, innerJoinResult) => {
+      if (innerJoinErr) {
+        console.log(innerJoinErr);
+        return innerJoinErr;
+      }
+      console.log(innerJoinResult.rows);
+      const species = innerJoinResult.rows;
+      res.render('species-index', { species });
+    });
+  });
+}
+
+function renderAllSpecies(req, res) {
+  const speciesQuery = 'SELECT * FROM species';
+  pool.query(speciesQuery, (err, result) => {
+    if (err) {
+      console.log(err);
+      return err;
+    }
+    const speciesObjArr = result.rows;
+    console.log(speciesObjArr);
+    res.render('species-all', { speciesObjArr });
+  });
+}
+
+function editSpecies(req, res) {
+  const id = req.params.index;
+  const speciesQuery = `SELECT * FROM species WHERE id = '${id}'`;
+  pool.query(speciesQuery, (err, result) => {
+    if (err) {
+      console.log(err);
+      return;
+    }
+    const speciesToEdit = result.rows[0];
+    res.render('species-edit', { speciesToEdit });
+  });
+}
+
+function speciesEditData(req, res) {
+  const { speciesId, speciesName, scientificName } = req.body;
+  const speciesQuery = `UPDATE species SET name = '${speciesName}', scientific_name = '${scientificName}' WHERE id ='${speciesId}'`;
+  pool.query(speciesQuery, (err, result) => {
+    if (err) {
+      console.log(err);
+      return;
+    }
+    console.log(result);
+    res.redirect(`/species/${speciesId}`);
+  });
+}
+
+function deleteSpecies(req, res) {
+  const deleteSpeciesId = req.params.index;
+  const speciesQuery = `DELETE FROM species WHERE id = '${deleteSpeciesId}'`;
+  pool.query(speciesQuery, (err, result) => {
+    if (err) {
+      console.log(err);
+      return;
+    }
+    console.log(result);
+    res.redirect('/species/all');
+  });
+}
+
+function handleComments(req, res) {
+  const { id } = req.params;
+  const { comment } = req.body;
+  const postedBy = req.cookies.userName;
+  const query = `INSERT INTO comments (comment_text, note_id,posted_by) VALUES ('${comment}', '${id}', '${postedBy}')`;
+  pool.query(query, (err, result) => {
+    if (err) {
+      console.log(`insert err${err}`);
+      return;
+    }
+    console.log(`insert success in comments table ${result}`);
+    res.redirect('/');
+  });
+}
+// get routes
+// renders a form which creates a new note.
+app.get('/note', createNewNote);
```

Comment:
Great work separating callback functions above, this makes your route logic easier to read here